test/unit-test/sigv4_utest.c

```diff
@@ -64,11 +209,184 @@ void formatAndVerifyInputDate( const char * pInputDate,
     tearDown();
 }
 
+/*==================== OpenSSL Based implementation of Crypto Interface ===================== */
+
+static int32_t valid_sha256_init( void * pHashContext )
+{
+    if( SHA256_Init( ( SHA256_CTX * ) pHashContext ) == 1 )
+    {
+        return 0;
+    }
+
+    return -1;
+}
+
+static int32_t valid_sha256_update( void * pHashContext,
+                                    const uint8_t * pInput,
+                                    size_t inputLen )
+{
+    if( SHA256_Update( ( SHA256_CTX * ) pHashContext, pInput, inputLen ) )
+    {
+        return 0;
+    }
+
+    return -1;
+}
+
+static int32_t valid_sha256_final( void * pHashContext,
+                                   uint8_t * pOutput,
+                                   size_t outputLen )
+{
+    if( SHA256_Final( ( uint8_t * ) pOutput, ( SHA256_CTX * ) pHashContext ) )
+    {
+        return 0;
+    }
+
+    return -1;
+}
+
+/*==================== Echo Implementation of Crypto Interface ===================== */
+
+static uint8_t hashEchoBuffer[ SIGV4_HASH_MAX_BLOCK_LENGTH ];
+static size_t hashInputLen;
+
+/* These hash functions simply take the input and write it back to the output.
+ * The purpose of which is make it possible to write tests without having to
+ * know the computed hash of the string to sign. */
+static int32_t echo_hash_init( void * pHashContext )
+{
+    return 0;
+}
+
+static int32_t echo_hash_update( void * pHashContext,
+                                 const uint8_t * pInput,
+                                 size_t inputLen )
+{
+    hashInputLen = inputLen;
+    ( void ) memcpy( hashEchoBuffer, pInput, inputLen );
+}
+
+static int32_t echo_hash_final( void * pHashContext,
+                                uint8_t * pOutput,
+                                size_t outputLen )
+{
+    ( void ) memcpy( pOutput, hashEchoBuffer, hashInputLen );
+}
+
+/*==================== Failable Implementation of Crypto Interface ===================== */
+
+#define HAPPY_PATH_HASH_ITERATIONS    12U
```

Comment:
This doesn't sound like an appropriate name to use?

Reply:
Yes, updated the name